backend/PythonClient/multirotor/monitor/point_deviation_monitor.py

```diff
@@ -125,21 +131,35 @@ def calculate_actual_distance(self):
 
     def draw_trace_3d(self):
         graph_dir = self.get_graph_dir()
+
         if not self.breach_flag:
             title = f"{self.target_drone} Planned vs. Actual\nDrone speed: {self.mission.speed} m/s\nWind: {self.wind_speed_text}"
         else:
             title = f"(FAILED) {self.target_drone} Planned vs. Actual\nDrone speed: {self.mission.speed} m/s\nWind: {self.wind_speed_text}"
-        grapher = ThreeDimensionalGrapher()
-        grapher.draw_trace_vs_planned(planed_position_list=self.mission.points,
+
+        self.grapher.draw_trace_vs_planned(planed_position_list=self.mission.points,
                                       actual_position_list=self.est_position_array,
                                       full_target_directory=graph_dir,
                                       drone_name=self.target_drone,
-                                      title=title
-                                      )
+                                      title=title)
 
-        grapher.draw_interactive_trace_vs_planned(planed_position_list=self.mission.points,
+        interactive_html_content = self.grapher.draw_interactive_trace_vs_planned(planed_position_list=self.mission.points,
                                                   actual_position_list=self.est_position_array,
                                                   full_target_directory=graph_dir,
                                                   drone_name=self.target_drone,
-                                                  title=title
-                                                  )
+                                                  title=title)
+        #Generate timestamp
+        timestamp = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+
+        # Upload the HTML file directly to GCS
+        file_name = f"html_reports/{self.target_drone}/interactive/{self.target_drone}_{timestamp}_interactive.html"
+        #file_name = f"{self.target_drone}/{self.target_drone}_interactive.html"    
+        with lock:
+            try:        
+                self.air_sim_app.upload_to_gcs(file_name, interactive_html_content)
+                print(f"Html Report successfully uploaded to {file_name} in GCS.")
+            except Exception as e:
+                print(f"Failed to upload to GCS. Error: {str(e)}")
```

Comment:
This does not align with our folder structure. I'm not even sure if this is needed since the structure is already defined in `simulation_task_manager.py`. It would be better to just leverage that.